app/src/main/java/com/cdio/solitaire/imageanalysis/SolitaireAnalysisModel.java

```diff
@@ -0,0 +1,223 @@
+package com.cdio.solitaire.imageanalysis;
+
+import static org.opencv.imgproc.Imgproc.ADAPTIVE_THRESH_MEAN_C;
+import static org.opencv.imgproc.Imgproc.THRESH_BINARY;
+import static java.lang.Double.max;
+import static java.lang.Double.min;
+
+import android.graphics.Bitmap;
+
+import org.opencv.android.Utils;
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.MatOfPoint2f;
+import org.opencv.core.Point;
+import org.opencv.core.Rect;
+import org.opencv.core.RotatedRect;
+import org.opencv.core.Size;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SolitaireAnalysisModel {
+
+    // Load in OpenCV library
+    static {
+        System.loadLibrary("opencv_java4");
```

Comment:
As said last comment, since you use this multiple places, maybe just do it in main?

Reply:
Now moved to mainActivity